sklearn/feature_extraction/dict_vectorizer.py

```diff
@@ -162,9 +165,14 @@ def _transform(self, X, fitting):
 
         # collect all the possible feature names and build sparse matrix at
         # same time
+        if self.drop_first_category:
+            to_drop = {}
         for x in X:
             for f, v in six.iteritems(x):
                 if isinstance(v, six.string_types):
+                    if self.drop_first_category and f not in to_drop:
```

Comment:
Hi Joel,

I like your solution! I've intentionally avoided splitting the string using a separator to overcome issues of ambiguity ― I hope people don't ever use the `=` character in columns names, but you never know :-) Let me know if you want me to implement your suggestion, and I'll update my PR.

> I also fear this is too sensitive to the ordering of the data for the user to find it explicable.

That's a valid point. In my own project to overcome such problem, I've stored the dictionaries that I want to pass to `DictVectorizer` inside a "master" dictionary. This master dictionary has keys that can be sorted in a way that guarantees the first category is deterministic.

```x = vectorizer.fit_transform([v for k, v in sorted(master.items())])```

In this example a key in master could be something like the following tuple:

```(582498109, 'Desktop')```

... where `Desktop` is the level I want to drop, and each id (the first element in the tuple) is associated with multiple devices, such as `Tablet`, `Mobile`, etc. I appreciate this is specific to my use case and not always true.

To be entirely fair, as a Data Scientist, 99% of the times you don't really care about which category is being dropped since that is simply your baseline. I guess in those situations when you need a specific category to be dropped, you can always build your own sorting function to pass to the `key` argument in `sorted`.

What do you think?